tensorflow/python/kernel_tests/decode_raw_op_test.py

```diff
@@ -20,14 +20,17 @@
 from __future__ import print_function
 
 import tensorflow as tf
-
+import sys
 
 class DecodeRawOpTest(tf.test.TestCase):
 
   def testToUint8(self):
     with self.test_session():
       in_bytes = tf.placeholder(tf.string, shape=[2])
-      decode = tf.decode_raw(in_bytes, out_type=tf.uint8)
+      if sys.byteorder == "big":
+        decode = tf.decode_raw(in_bytes, out_type=tf.uint8, little_endian=False)
```

Comment:
i.e. suppose we have a byte string that is encoded little endian
```const* foo_little[]="Aa";```
then it will be encoded as big-endian as
```const* foo_big[]="aA";```
Then on a big-endian or little-endian machine big-endian data can be converted to uint16 array as
```c++
std::vector<uint16> vals;
for(char* p=foo_little; *p != 0; p+=2)
   vals.push_back((p[0]<<8)  | (p[1]));
}
```
Little endian data can be converted to a big-endian or ltitle-endian machine with
```c++
std::vector<uint16> vals;
for(char* p=foo_big; *p != 0; p+=2)
   vals.push_back((p[0])  | (p[1]<<8));
}
```
So, when you construct the string in python you are saying it is little endian or big endian... so the test should look like

```python
big_endian_data = "aA"
little_endian_data = "Aa"
a=tf.decode_raw(little_endian_data, little_endian=None);
b=tf.decode_raw(little_endian_data, little_endian=True);
c=tf.decode_raw(big_endian_data, little_endian=False);
assertEqual(a,b)
assertEqual(a,c)
manual_decode =  little_endian_data[0] | (little_endian_data[1]<<8)
assertEqual(a, manual_decode) # this verifies that endianness is correct
```
(note this is for unsigned data, since the number encode is less than 1<<15 we can use this conversion routine for these numbers, but you need to adapt this for signed values.